persist/fs/seek_manager_test.go

```diff
@@ -58,3 +66,214 @@ func TestSeekerManagerCacheShardIndices(t *testing.T) {
+		tickCh <- struct{}{}
+	}
+
+	metadata := testNs1Metadata(t)
+	seekers := []ConcurrentFileSetSeeker{}
+
+	require.NoError(t, m.Open(metadata))
+	// Steps is a series of steps for the test. It is guaranteed that at least
+	// one (not exactly one!) tick of the openCloseLoop will occur between every step.
+	steps := []struct {
+		title string
+		step  func()
+	}{
+		{
+			title: "Make sure it didn't clean up the seekers which are still in retention",
+			step: func() {
+				m.RLock()
+				for _, shard := range shards {
+					require.Equal(t, 1, len(m.seekersByTime(shard).seekers[startNano].seekers))
+				}
+				m.RUnlock()
+			},
+		},
+		{
+			title: "Borrow a seeker from each shard and then modify the clock such that they're out of retention",
+			step: func() {
+				for _, shard := range shards {
+					seeker, err := m.Borrow(shard, now)
+					require.NoError(t, err)
+					require.NotNil(t, seeker)
+					seekers = append(seekers, seeker)
+				}
+
+				fakeTimeLock.Lock()
+				fakeTime = fakeTime.Add(10 * metadata.Options().RetentionOptions().RetentionPeriod())
+				fakeTimeLock.Unlock()
+			},
+		},
+		{
+			title: "Make sure the seeker manager cant be closed while seekers are borrowed",
+			step: func() {
+				require.Equal(t, errCantCloseSeekerManagerWhileSeekersAreBorrowed, m.Close())
+			},
+		},
+		{
+			title: "Make sure that none of the seekers were cleaned up during the openCloseLoop tick (because they're still borrowed)",
+			step: func() {
+				m.RLock()
+				for _, shard := range shards {
+					require.Equal(t, 1, len(m.seekersByTime(shard).seekers[startNano].seekers))
+				}
+				m.RUnlock()
+			},
+		},
+		{
+			title: "Return the borrowed seekers",
+			step: func() {
+				for i, seeker := range seekers {
+					require.NoError(t, m.Return(shards[i], now, seeker))
+				}
+			},
+		},
+		{
+			title: "Make sure that the returned seekers were cleaned up during the openCloseLoop tick",
+			step: func() {
+				m.RLock()
+				for _, shard := range shards {
+					byTime := m.seekersByTime(shard)
+					byTime.RLock()
+					_, ok := byTime.seekers[startNano]
+					byTime.RUnlock()
+					require.False(t, ok)
+				}
+				m.RUnlock()
+			},
+		},
+	}
+
+	for _, step := range steps {
+		// Wait for two notifications between steps to guarantee that the entirety
+		// of the openCloseLoop is executed at least once
+		<-tickCh
+		<-tickCh
+		step.step()
+	}
+
+	// Background goroutine that will pull notifications off the tickCh so that
+	// the openCloseLoop is not blocked when we call Close()
+	go func() {
+		for {
+			select {
+			case <-tickCh:
+				continue
+			case <-cleanupCh:
+				return
+			}
+		}
+	}()
+
+	// Restore previous interval once the openCloseLoop ends
+	require.NoError(t, m.Close())
+	// Make sure there are no goroutines still trying to write into the tickCh
+	// to prevent the test itself from interfering with the goroutine leak test
+	close(cleanupCh)
+	seekManagerCloseInterval = oldInterval
```

Comment:
Might be worth doing the restore of this in a defer at the top so at least that logic sticks together (and one doesn't get deleted without the other).

Reply:
I actually think this is completely unnecessary now that I'm using a custom sleepFn, just gonna make it a constant again